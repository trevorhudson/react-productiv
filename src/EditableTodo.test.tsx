import { render,fireEvent,getByLabelText, queryByText,queryAllByText } from "@testing-library/react";
import EditableToDo from './EditableTodo';
import React from "react";

describe("EditableToDo component test", function() {
  const todo =
    {
      id: "1",
      title: 'todo1',
      description: 'desc1',
      priority: "3"
    };

  it("renders", function() {
    render(<EditableToDo todo={todo} update = {()=>{}} remove = {()=>{}} />);
  });
  it("matches snapshot", function() {
    const {container} =     render(<EditableToDo todo={todo} update = {()=>{}} remove = {()=>{}} />);
    expect(container).toMatchSnapshot();
  });
  it("todos data and GO feature features to be there", function() {
    const result =  render(<EditableToDo todo={todo} update = {()=>{}} remove = {()=>{}} />);
    expect(result.queryByText("todo1")).toBeInTheDocument();
    expect(result.queryByText("desc1")).toBeInTheDocument();
  });
  it("delete button works", function() {
    const removeMock = jest.fn();
    removeMock.mockClear();
    const { container } = render(<EditableToDo remove={removeMock} todo={todo} update = {()=>{}} />);
    const delBtn = container.querySelector(".EditableTodo-delBtn")!;
    fireEvent.click(delBtn);
    expect(removeMock).toHaveBeenCalledTimes(1);
  })
  it("edit button works", function() {
    const { container } = render(<EditableToDo todo={todo}  update = {()=>{}} remove = {()=>{}}/>);
    const editBtn = container.querySelector(".EditableTodo-toggle")!;
    expect(container.querySelector(".NewTodoForm")).not.toBeInTheDocument();
    fireEvent.click(editBtn);
    expect(container.querySelector(".NewTodoForm")).toBeInTheDocument();
  })
});