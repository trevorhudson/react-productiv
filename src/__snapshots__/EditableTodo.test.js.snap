// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableToDo component test matches snapshot 1`] = `
<div>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3"
    >
      <div
        class="float-end text-sm-end"
      >
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
      <div
        class="Todo"
      >
        <div>
          <b>
            todo1
          </b>
           
          <small>
            3
          </small>
        </div>
        <div>
          <small>
            desc1
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
